sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -1116,51 +1110,12 @@ public Write withMaxOutstandingBytes(long bytes) {
      * always enabled on batch writes and limits the number of outstanding requests to the Bigtable
      * server.
      *
-     * <p>When enabled, will also set default {@link #withThrottlingReportTargetMs} to 1 minute.
-     * This enables runner react with increased latency in flush call due to flow control.
-     *
      * <p>Does not modify this object.
      */
     public Write withFlowControl(boolean enableFlowControl) {
-      BigtableWriteOptions options = getBigtableWriteOptions();
-      BigtableWriteOptions.Builder builder = options.toBuilder().setFlowControl(enableFlowControl);
-      if (enableFlowControl) {
-        builder = builder.setThrottlingReportTargetMs(60_000);
-      }
-      return toBuilder().setBigtableWriteOptions(builder.build()).build();
-    }
-
-    /**
-     * Returns a new {@link BigtableIO.Write} with client side latency based throttling enabled.
-     *
-     * <p>Will also set {@link #withThrottlingReportTargetMs} to the same value.
-     */
-    public Write withThrottlingTargetMs(int throttlingTargetMs) {
-      BigtableWriteOptions options = getBigtableWriteOptions();
-      return toBuilder()
-          .setBigtableWriteOptions(
-              options
-                  .toBuilder()
-                  .setThrottlingTargetMs(throttlingTargetMs)
-                  .setThrottlingReportTargetMs(throttlingTargetMs)
-                  .build())
-          .build();
-    }
-
-    /**
-     * Returns a new {@link BigtableIO.Write} with throttling time reporting enabled. When write
-     * request latency exceeded the set value, the amount greater than the target will be considered
-     * as throttling time and report back to runner.
-     *
-     * <p>If not set, defaults to 3 min for completed batch request. Client side flowing control
-     * configurations (e.g. {@link #withFlowControl}, {@link #withThrottlingTargetMs} will adjust
-     * the default value accordingly. Set to 0 to disable throttling time reporting.
-     */
-    public Write withThrottlingReportTargetMs(int throttlingReportTargetMs) {
```

Comment:
same as above

Reply:
Please dont throw an exception, instead log a warning. We dont want to be in position where are user started using this flag in 2.59 and then their jobs start failing when they upgrade to 2.60. In other words throwing an UnsupportedOperationException is effectively the same as breaking binary compatibility